hydra-node/test/Hydra/HeadLogicSpec.hs

```diff
@@ -562,32 +573,39 @@ data StepState tx = StepState
   }
 
 -- | Asserts that the update function will update the state (return a NewState) for this Event
-step :: (MonadState (StepState tx) m, HasCallStack, IsChainState tx) => Event tx -> m (HeadState tx)
+step ::
+  (MonadState (StepState tx) m, HasCallStack, IsChainState tx, MonadCatch m) =>
+  Event tx ->
+  m (HeadState tx)
 step event = do
-  StepState{ headState, env, ledger} <- get
+  StepState{headState, env, ledger} <- get
   headState' <- assertNewState $ update env ledger headState event
-  put StepState{ env, ledger, headState = headState' }
+  put StepState{env, ledger, headState = headState'}
   pure headState'
 
 assertNewState ::
-  (HasCallStack, IsChainState tx, Applicative m) =>
+  (HasCallStack, IsChainState tx, MonadCatch m) =>
   Outcome tx ->
   m (HeadState tx)
 assertNewState = \case
-  NewState st _ -> pure st
-  OnlyEffects effects -> Hydra.Prelude.error $ "Unexpected 'OnlyEffects' outcome: " <> show effects
-  Error e -> Hydra.Prelude.error $ "Unexpected 'Error' outcome: " <> show e
-  Wait r -> Hydra.Prelude.error $ "Unexpected 'Wait' outcome with reason: " <> show r
+  NewState st -> pure st
+  Effects effects -> failure $ "Unexpected 'OnlyEffects' outcome: " <> show effects
+  Error e -> failure $ "Unexpected 'Error' outcome: " <> show e
+  Wait r -> failure $ "Unexpected 'Wait' outcome with reason: " <> show r
```

Comment:
Consider keeping using Error here as we do have the call stack. This would remove dependency to `transformers`